packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -496,8 +523,16 @@ export class CollabSpacesRuntime
 			) => {
 				for (let row = rowStart; row < rowStart + rowCount; row++) {
 					for (let col = colStart; col < colStart + colCount; col++) {
-						// -1 due to first row & col tracking IDs
-						if (row !== 0 && col !== 0) {
+						if (row === 0 || col === 0) {
+							// TBD(Pri1) : Today we will raise "column added" event before we have an ID, and thus exposing incorrect state of matrix to consumers.
+							// It's better if we do something like that:
+							// - suppress col/row addition events
+							// - raise them here, when we learn about new ID.
+							// It's not that simple, as proposed sequence would only work for a single row/col addition, not for many rows/columns added, but I'm sure we can figure out something.
+							// Alternatively, we need better APIs from matrix...
+							this.updateReverseMap(row - 1, col - 1);
```

Comment:
This is probably fine, but API looks a bit weird. I.e. we pass -1 for one of the parameters and that tells it what to update.
Maybe better to have that logic here and call two different APIs (updateColReverseMap, updateRowReverseMap) or single API (updateMap(what: "rows" | "cols", index).